### ISSUE TITLE
[FR] Add support for datasets which comprise pairs of images

### ISSUE THREAD

**Issue by robmarkcole** [voxel51/fiftyone#1298]
### Proposal Summary
There are applications such as super resolution or image feature matching where the unit of data is a pair (or even triplet) of images & annotations. The scope of this FR is to support these datasets. 

Note: I believe this may be on your backlog but could not locate an issue. I guess a quick hack would be to join the image pair into a single image

### What areas of FiftyOne does this feature affect?

-   [ ] App: FiftyOne application
-   [ ] Core: Core `fiftyone` Python library
-   [ ] Server: FiftyOne server

### Details
- As an example dataset I link a thermal/rgb pair: https://www.flir.co.uk/oem/adas/adas-dataset-form/
- Also a super res dataset: https://github.com/JasonBoy1/TextZoom

### Willingness to contribute
-   [ ] Yes. I can contribute this feature independently.
-   [ ] Yes. I would be willing to contribute this feature with guidance from
        the FiftyOne community.
-   [x] No. I cannot contribute this feature at this time.

**Comment by jasoncorso** [comment]
@robmarkcole Yes, we agree this is a relevant and interesting feature.  In fact, being able to visually compare pairs or sets of images as well as other datatypes, such as detections, videos, etc. is relevant.  This has come up a few times. I'd say we're in the ideation stage of defining constraints and requirements for it.  So, as many details as you can provide would helpful.  As many details as any here reading this can provide will be helpful!

**Comment by robmarkcole** [comment]
I think an MVP would be side-by-side image display. Alternatively you could overlay them and use a slider like here https://github.com/robmarkcole/streamlit-image-juxtapose
